payment-handler/can-make-payment-event.https.html

```diff
@@ -0,0 +1,144 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>Tests for CanMakePaymentEvent</title>
+<link rel="help" href="https://w3c.github.io/payment-handler/#the-canmakepaymentevent">
+<link rel="manifest" href="manifest.json">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+async function registerAppAndBuildPaymentRequest(methodName) {
+  await navigator.serviceWorker.register("app-can-make-payment.js");
+  const registration = await navigator.serviceWorker.ready;
+  await registration.paymentManager.instruments.set("instrument-key", {
+    name: "Test Payment Method",
+    enabledMethods: [methodName],
+  });
+  const request = new PaymentRequest(
+    [
+      {
+        supportedMethods: methodName,
+      },
+    ],
+    {
+      total: {
+        label: "Total",
+        amount: {
+          currency: "USD",
+          value: "0",
+        },
+      },
+    }
+  );
+  return request;
+}
+
+promise_test(async t => {
+  const methodName = window.location.origin + "/canMakePayment-false";
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(
+      err.name,
+      "NotAllowedError",
+      "If it throws, then it must be NotAllowedError"
+    );
+    paymentRequestCanMakePaymentResult = false;
```

Comment:
I'm still not super comfortable with this, tbh. I think we should do the assert_equals for the Error, but not `paymentRequestCanMakePaymentResult`.

This should just error at `assert_false`.

Reply:
Let's discuss with more bandwidth at the upcoming editors meeting.